src/bm_sim/P4Objects.cpp

```diff
@@ -3236,6 +3337,60 @@ P4Objects::change_conditional_next_node_rt(const std::string &pipeline_name,
     modify_json_value(pipeline_name+" conditional", src_conditional_name, edge_name, dst_node_name);
   }
 }
+
+void
+P4Objects::change_register_array_size_rt(const std::string& name, 
+                                         const std::string& register_array_size) {
+  RegisterArray* register_array = get_register_array(name);
+  int new_register_array_size = std::stoi(register_array_size);
+  int ori_register_array_size = register_array->size();
+  if (new_register_array_size == ori_register_array_size) {
+    return;
+  } else if (new_register_array_size > ori_register_array_size) {
+    std::vector<Register> registers_new;
+    registers_new.reserve(new_register_array_size);
+    auto lock = register_array->unique_lock();
+    int i = 0;
+    while (i < new_register_array_size) {
+      if (i < ori_register_array_size) {
+        registers_new.push_back(register_array->at(i));
+      } else {
+        registers_new.emplace_back(register_array->get_bitwidth(), register_array);
```

Comment:
What does this do? 

BTW, the default value for an uninitialized register is 0.